src/com/TigersIter2/managers/AreaEffectManager.java

```diff
@@ -11,19 +14,38 @@
  */
 public class AreaEffectManager {
 
-    private Entity entity;
+    private Entity entityOnTile;
     private AreaEffect areaEffect;
-    private StatsModifier statsMod;
     private InstantDeath instantDeath;
+    private TakeDamage takeDamage;
+    private HealDamage healDamage;
+    private LevelUp levelUp;
 
 
-    public void getStatsModifier(){
+    public AreaEffectManager(Entity entity, AreaEffect effectType, Location l){
+        entityOnTile = entity;
+        areaEffect = effectType;
+        areaEffect.setLocation(l);
+        instantDeath = new InstantDeath();
+        takeDamage = new TakeDamage();
+        healDamage = new HealDamage();
+        levelUp = new LevelUp();
+    }
 
-        switch(getAreaEffect()) {
+    public void affectEntityOnTile(){
 
-            case "instantDeath": statsMod = instantDeath.affectEntity();
+        switch(getAreaEffect()) {
 
+            case "instantDeath": instantDeath.affectEntity(entityOnTile);
+                break;
+            case "takeDamage": takeDamage.affectEntity(entityOnTile);
+                break;
+            case "healDamage": healDamage.affectEntity(entityOnTile);
+                break;
+            case "levelUp": levelUp.affectEntity(entityOnTile);
+                break;
                 // add other cases for other area effects
+                // is teleport an area affect???
```

Comment:
Why is this not done is such a way that it's possible to `do getAreaEffect().afffectEntity(entity)`?